mne/preprocessing/eog.py

```
@@ -249,3 +250,112 @@ def create_eog_epochs(raw, ch_name=None, event_id=998, picks=None, tmin=-0.5,
+        The version of the data with the EOG removed.
+    weights : ndarray, shape (n_eog_channels, n_channels)
+        The regression weights. For each EOG channel, the fitted regression
+        weight to each data channel. The ordering of the weights matches the
+        ordering of the channels of the object given as ``inst`` parameter.
+    intercept : ndarray, shape (n_channels,)
+        For each data channel, the fitted intercept. The ordering of the
+        intercepts matches the ordering of the channels of the object given as
+        ``inst`` parameter.
+
+    References
+    ----------
+    .. [1] Gratton, G. and Coles, M. G. H. and Donchin, E. (1983). A new method
+           for off-line removal of ocular artifact. Electroencephalography and
+           Clinical Neurophysiology, 468-484.
+           https://doi.org/10.1016/0013-4694(83)90135-9
+    .. [2] Croft, R. J. and Barry, R. J. (1998). EOG correction: a new
+           aligned-artifact average solution. Clinical Neurophysiology, 107(6),
+           395-401. http://doi.org/10.1016/s0013-4694(98)00087-x
+    """
+    # Handle defaults for EOG channels parameter
+    if eog_channels is None:
+        eog_picks = pick_types(inst.info, meg=False, ref_meg=False, eog=True)
+        eog_channels = [inst.ch_names[ch] for ch in eog_picks]
+    elif isinstance(eog_channels, str):
+        eog_channels = [eog_channels]
+    eog_picks = [inst.ch_names.index(ch) for ch in eog_channels]
+    if len(eog_picks) == 0:
+        raise RuntimeError('No EOG channels found in given data instance. '
+                           'Make sure channel types are marked properly.')
+
+    # Default picks: channels to remove EOG from
+    if picks is None:
+        picks = pick_types(inst.info, meg=False, ref_meg=False, eeg=True)
+
+    # This is the data from which the EOG should be removed. When operating on
+    # epochs, concatenate all the epochs into a channels x time matrix.
+    data = inst._data
```

this attribute is only present for preloaded data. Also I would not modify the input inst or introduce a copy param that defaults to false. If true do:

`inst = inst.copy().load_data()`

and inst._data will be there.

---

Better than this I think is _check_preload. We don't want to automatically preload data like this, it can kill users systems due to memory requirements. Better to raise an error if not preloaded, which IIRC is what _check_preload does